### PR TITLE
Add test project for admin, migrations, and tests

### PR DESCRIPTION
**Pull request by adam-thomas**
`pip install -e`-ing libraries in order to make migrations or test the admin is a hassle, and can lead to some awkward versioning problems.  To avoid having to do the install/uninstall dance every time, I've added a barebones project that can be used for both that and running tests.

@incuna/backend review please?

### CODE REVIEW

**Comment by adam-thomas** [comment]
Bump?

File: Makefile [hunk 1]
```diff
@@ -12,6 +15,18 @@ release:
 	@python setup.py register sdist bdist_wheel upload
 
 test:
-	@coverage run groups/tests/run.py --verbosity=${VERBOSITY}
+	@coverage run test_project/manage.py test groups --verbosity=${VERBOSITY}
 	@flake8 .
-	@coverage report --fail-under=100
+	@DJANGO_SETTINGS_MODULE=test_project.settings coverage report --fail-under=100
```
Comment (meshy) [inline Makefile]: `--fail-under` is probably best defined in the `setup.cfg`:

```
[coverage:report]
fail-under=100
```

Comment (adam-thomas) [inline Makefile]: neat, cheers!

**Comment by adam-thomas** [comment]
updated :)

File: test_project/settings.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    'django.contrib.contenttypes',
+    'django.contrib.auth',
+    'django.contrib.admin',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+)
+TEMPLATE_CONTEXT_PROCESSORS = TEMPLATE_CONTEXT_PROCESSORS + (
+    'django.core.context_processors.request',
+)
```
Comment (meshy) [inline test_project/settings.py]: Might as well use the [new `TEMPLATES` setting](https://docs.djangoproject.com/en/1.8/ref/templates/upgrading/).

Comment (adam-thomas) [inline test_project/settings.py]: What's `DIRS` supposed to include?  These instructions are missing a whole bunch of stuff.

Comment (meshy) [inline test_project/settings.py]: `DIRS` can be excluded. It [defaults to `[]`](https://docs.djangoproject.com/en/1.8/ref/settings/#templates).

Comment (adam-thomas) [inline test_project/settings.py]: OK thanks :)

**Comment by meshy** [comment]
LGTM

**Comment by adam-thomas** [comment]
Thanks :)

**Comment by adam-thomas** [comment]
IMO, @incuna/backend, it'd be useful to do this on most or all of our libraries that contain models, especially if there's any admin work involved.

**Pull request merged**